tpot/base.py

```diff
@@ -392,27 +392,17 @@ def fit(self, features, classes, sample_weight=None):
         """
         features = features.astype(np.float64)
 
-        # Check that the input data is formatted correctly for scikit-learn
-        if self.classification:
-            clf = DecisionTreeClassifier(max_depth=5)
-        else:
-            clf = DecisionTreeRegressor(max_depth=5)
+        if np.any(np.isnan(features)):
+            features = self._impute_values(features)
 
-        try:
-            clf = clf.fit(features, classes)
-        except Exception:
-            raise ValueError('Error: Input data is not in a valid format. '
-                             'Please confirm that the input data is scikit-learn compatible. '
-                             'For example, the features must be a 2-D array and target labels '
-                             'must be a 1-D array.')
+        self._check_dataset(features, classes)
```

Comment:
why not use [check_X_y](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/validation.py#L435)? it can be imported by `from sklearn.utils import check_X_y`

Another benefits from `check_X_y` is that the return value of y are converted to `np.float64` by [this line](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/validation.py#L532). So `features = features.astype(np.float64)` in self.fit() could be deleted then.